internal/cache/persistent_lru_cache.go

```diff
@@ -359,12 +322,7 @@ func (c *PersistentCache) listCacheCleanupLocked(ctx context.Context) {
 			break
 		}
 
-		// unlock before the expensive operation
-		c.listCacheMutex.Unlock()
-		delerr := c.cacheStorage.DeleteBlob(ctx, oldest.BlobID)
-		c.listCacheMutex.Lock()
-
-		if delerr != nil {
+		if delerr := c.cacheStorage.DeleteBlob(ctx, oldest.BlobID); delerr != nil {
```

Comment:
This will serialize all deletes. The existing unlock+lock mechanism allowed N parallel deletes to get spread over the workpool.